packages/gatsby/src/query/file-parser.js

```diff
@@ -366,6 +366,9 @@ async function findGraphQLTags(
         // Look for exported page queries
         traverse(ast, {
           ExportNamedDeclaration(path, state) {
+            if (path.node.source) {
```

Comment:
Technically the property has to be `null` if there was no `from`. I guess in practice it doesn't matter much.